sonar-java-plugin/src/main/resources/static/documentation.md

```diff
@@ -63,6 +63,19 @@ Example: `sonar.java.source=1.6`
 
 If the property is provided, the analysis will take the source version into account, and execute related rules accordingly. At run time, each of these rules will be executed – or not – depending of the Java version used by sources within the project. For instance, on a correctly configured project built with Java 6, rules targeting Java 7 and Java 8 will never raise issues, even though they are enabled in the associated rule profile.
 
+## Skipping unchanged files
+Starting with SonarQube 9.4, and by default, the Java analyzer optimizes the analysis of unchanged files in pull requests.
+In practice, this means that on a file that has not changed, the analyzer only runs a restricted list of checks.
+To get a better understanding of the rule exclusion mechanism, keep in mind that:
+* Rules that need to run on multiple files to decide whether they need to raise issues are always executed
+* Rules that need to run at the end of the analysis to decide whether they need to raise issues are always executed
+* Rules that are defined out of the `sonar.java.checks` package are always executed
+
+This last criteria implies that checks defined in custom plugins cannot be skipped.
```

Comment:
I would replace that by:

"This last criteria implied that [custom rules] cannot be skipped". Where [custom rules] is a link to the doc about how to create custom rules.